File: src/Ex. (85).m
a = [5, 2, 8, 4, 6, 22, 8, 9];
m = 15;
t = '';

maxVal = max(a);
n = length(a);

for i = 1:n
    f = floor((m / maxVal) * a(i));
    
    for k = 1:m
        if k <= f
            t = [t '▮'];
        else
            t = [t '▯'];
        end
    end
    
    t = [t '\n'];
end

fprintf(t);

% Ex. (85) - Horizontal chart with UTF characters proportional with max array.

% References
% Paul A. Gagniuc. Coding Examples from Simple to Complex - Applications in Python, Springer, 2024, pp. 1-245.
% Paul A. Gagniuc. Coding Examples from Simple to Complex - Applications in MATLAB, Springer, 2024, pp. 1-255.
% Paul A. Gagniuc. Coding Examples from Simple to Complex - Applications in Javascript, Springer, 2024, pp. 1-240.

File: src/Ex. (85).js
let a = [5, 2, 8, 4, 6, 22, 8, 9];

let m = 15;
let t = '';
let max = 0;

let n = a.length;

for(let i=0; i<n; i++){
    if(a[i]>max){max=a[i];}
}

for(let i=0; i<n; i++){
    
    f = Math.floor((m/max) * a[i]);
    
    for(let k=0; k<m; k++){
        
        if(k<f){
            t += '▮';
        } else {
            t += '▯';
        }
    }

    t += '\n';
}

print(t);

// 5.1.45 Ex. (85) – Horizontal chart with UTF characters proportional with max array